src/pages/templates/[templateId].tsx

```diff
@@ -0,0 +1,764 @@
+import { Box, DarkMode, Flex, Image } from "@chakra-ui/react";
+import { CopyButton } from "components/homepage/AnimatedCLICommand/AnimatedCLICommand";
+import { ProductButton } from "components/product-pages/common/ProductButton";
+import { HomepageTopNav } from "components/product-pages/common/Topnav";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { PageId } from "page-id";
+import { TEMPLATE_DATA, TemplateCardProps } from "data/templates/templates";
+import React from "react";
+import {
+  Heading,
+  Link,
+  LinkButton,
+  Text,
+  TrackedLink,
+  TrackedLinkButton,
+} from "tw-components";
+import { ThirdwebNextPage } from "utils/types";
+import { getDisplayTagFromTagId } from "../../components/templates/Wrapper";
+import { MarkdownRenderer } from "components/contract-components/published-contract/markdown-renderer";
+import { NextSeo } from "next-seo";
+
+type TemplateContentsProps = {
+  overview: React.ReactNode;
+  featurePoints: React.ReactNode[];
+  runningLocally?: React.ReactNode;
+};
+
+const CodeBlock: React.FC<{ text: string }> = (props) => {
+  return (
+    <Flex
+      border="1px solid rgba(255, 255, 255, 0.2)"
+      borderRadius="md"
+      flexShrink={0}
+      py={3}
+      px={4}
+      my={4}
+      minW={"300px"}
+      gap={1}
+      align="center"
+      alignSelf="start"
+    >
+      <Text color="white" fontFamily="mono" fontSize="16px" fontWeight="500">
+        <span>{props.text}</span>
+      </Text>
+      <CopyButton text={props.text} />
+    </Flex>
+  );
+};
+
+const TemplateContents: React.FC<TemplateContentsProps> = (props) => {
+  return (
+    <>
+      <Heading as="h3" fontSize="24px" fontWeight={700} mt={12} mb={4}>
+        Overview
+      </Heading>
+
+      {props.overview}
+
+      {props.featurePoints.length > 0 && (
+        <>
+          <Box as="ul" listStyleType="inherit" pl={4} mt={4}>
+            {props.featurePoints.map((feature, idx) => (
+              <Box
+                as="li"
+                key={idx}
+                fontWeight={400}
+                fontSize={16}
+                lineHeight={1.5}
+                opacity={0.7}
+                color="whiteAlpha.900"
+                mb={2}
+              >
+                {feature}
+              </Box>
+            ))}
+          </Box>
+        </>
+      )}
+
+      <Heading as="h3" fontSize="24px" fontWeight={700} mt={12} mb={4}>
+        Running locally
+      </Heading>
+
+      <Box
+        fontWeight={400}
+        fontSize={16}
+        lineHeight={1.5}
+        opacity={0.7}
+        color="whiteAlpha.900"
+        mb={2}
+      >
+        {props.runningLocally}
+      </Box>
+    </>
+  );
+};
+
+// Use the "id" fields from the templates array of objects to create a mapping of contents for each template id
+const templateContents: Record<
```

Comment:
this is a lot of data to always load here, can we lazy-load these based on the templateId instead?

Reply:
@jnsdls hmm I thought this would be generated at build time anyway?